src/fetcher.ts

```diff
@@ -756,11 +759,39 @@ export default class Fetcher implements CallbackifyInterface {
     this.ns = getNS(this.store.rdfFactory)
     this.timeout = options.timeout || 30000
 
-    // solidFetcher is deprecated
-    this._fetch = options.fetch
+    // solidFetcher is deprecated, instead use solidFetch
+    const originalFetch = options.fetch
                || (typeof global !== 'undefined' && (global.solidFetcher || global.solidFetch))
                || (typeof window !== 'undefined' && (window.solidFetcher || window.solidFetch))
                || crossFetch
+    const inBrowserFetch = async (uri,options) => {
+      if(!uri.startsWith('ls://')) return originalFetch(uri,options);
+      const success =  {
+           ok:true,
+           status:200,
+           headers : {
+             "content-type":contentTypeLookup(uri),
+             "wac-allow":`user="read write append control",public="read"`,
+           },
+      }
+      const failure =  { ok:false, status:500 }
+      options.method ||= 'GET';
+      if(options.method==='PUT'){
+        window.localStorage.setItem(uri,options.body);
+        return new Response(null,success);
+      }
+      else if(options.method==='GET'){
+        let content = window.localStorage.getItem(uri);
+        if(content) return new Response(content,success);
+        else {
+          failure.status = 404;
+          return(failure);
+        }
+      }
+      else return(failure);
```

Comment:
Would it be too difficult to support PATCH? And what about DELETE?

I have to say that this may work most of the time, but it may create some unexpected behaviour (I guess that's ok if we document this as "experimental").

The way I do it in my apps, I'm not really working with Solid when I use localStorage, in a different layer I am serializing my data as JsonLD, and then I am writing or modifying plain javascript objects into localStorage. So my code dealing with localStorage is not aware that it's dealing with semantic data.

Reply:
I suppose this solution actually doesn't know it's semantic data either - and that's the problem with supporting PATCH, which would require parsing of e.g. a sparql update query and then modification of the document. That said, rdflib is probably pretty close to having the parsing functionality already, so it might be possible. Either way, it probably makes sense for PATCH to be implemented in a separate PR.
DELETE seems more manageable.
I don't fully understand the limitations on content or keys in local storage but I assume that's what you mean by potential for unexpected behaviour?

Reply:
Ha. Took me a while to work it out but we don't need to implement PATCH because update with a WAC and non http Uri calls updateLocalFile, which only uses a PUT
https://github.com/linkeddata/rdflib.js/blob/e3037a1d6816591900608784856ecde6afd5eba1/src/update-manager.ts#L993

Reply:
Perhaps the failure could provide a message, e.g. localStorage only supports GET and PUT. Use updater.update to patch the document.

Reply:
@josephguillaume  - the fact that update defaults to a PUT and that non-http URIs can be recognized as editable via wac-allow is due to my previous work to make rdflib compatible for file:// URLs.

@NoelDeMartin - Solid-rest does implement patch and container support and options and more.  At one point I had it working with indexedDb, Native File API, and Dropbox.  If someone needs a full-featured browser storage handler, they can dust off solid-rest and they should be good to go.  My thought was that all of that should be an outside library one could import and declare with the solidFetch global.  The purpose of this PR is to support rdflib's use of the storage - load, putBack, and update and minimal was my goal.

In Solid-Rest, I supported indexedDB and Native File Api through an external package called BrowserFS.  That means I could probably add support for them here with a line or two that would only be called if the user had BrowserFS imported.

Reply:
Although this PR does an end-run around PATCH, Solid-Rest does implement actual PATCH.  I'm not sure there is much difference between PUT and PATCH when it comes to local storage.

Reply:
> I don't fully understand the limitations on content or keys in local storage but I assume that's what you mean by potential for unexpected behaviour?

@josephguillaume Well what I mean with that is that someone may presume that using `lp://` will work the same way than using `http://`, but I'm sure there are many things that will not work. And if what works/doesn't work is not clear, than can make this library frustrating to work with.

I haven't worked a lot with the, so maybe it's not necessary to handle PATCH, I was just raising the point to clarify.

Reply:
> someone may presume that using `ls://` will work the same way than using `http://`, but I'm sure there are many things that will not work. And if what works/doesn't work is not clear, than can make this library frustrating to work with.

My intention is to have this clear distinction : ls:// (or more likely browser://localStorage/) can be used in rdflib and libraries that import it wtih the load, putBack, and update methods.  As far as those methods go, the only difference between browser: and http: would be in loading  containers AFAIK.  Do you know of something else?